azafea/tests/integration/__init__.py

```diff
@@ -33,17 +31,13 @@ def ensure_tables(self, *models):
                 dbsession.query(model).count()
 
     def ensure_no_tables(self):
+        connection = self.db._engine.connect()
```

Comment:
Can this object be used as a context manager, either directly or with `contextlib.closing()`?

Reply:
Amazing that this change has a 10% performance improvement…

Reply:
> Can this object be used as a context manager, either directly or with `contextlib.closing()`?

Of course. No idea why I always forget contexts…

> Amazing that this change has a 10% performance improvement…

Creating a connection for each table and for each test was really expensive :wink:.